[DEVEXP-1264] Upgrade PHP 7.1

juanariastn opened the pull request:
[DEVEXP-1264](https://tiendanube.atlassian.net/browse/DEVEXP-1264)

# Changes
- Update to `PHP 7.1`
- Update dependencies
- Add `rector` tool to analyse and detect PHP version updates
- Add README with helpful commands 👇 

# Helpful commands
## Run lint
```
docker compose -f docker-compose.cli.yml run --rm php-cs-fixer fix --verbose
```
## Run unit tests
```
docker compose run --no-deps --rm php-cli composer test
```
## Run rector
```
docker compose run --no-deps --rm php-rector php ../rector/vendor/bin/rector process
```

[DEVEXP-1264]: https://tiendanube.atlassian.net/browse/DEVEXP-1264?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

juanariastn committed:
chore: ignore files and new line at EOF

juanariastn committed:
Update PHP version to 7.1

This update includes a shift from PHP 7.0 to PHP 7.1 in composer.json and docker-compose.cli.yml and a significant change in the Dockerfile setup.
Additionally, composer.json and GitHub action files have been modified to comply with the new updates.

juanariastn committed:
Add Rector and associated Docker setup

This commit introduces `rector.php` to implement a type declaration rector rule, and adds a Dockerfile for `php-rector`. Additionally, a `php-rector` service is integrated into the `docker-compose.yml` file to ensure the proper Docker container setup.

juanariastn committed:
Fix lint

Add return type declarations to test methods

Minor adjustments were made to the class constants declaration in WebpOptimizer, making the constants public explicitly.

juanariastn committed:
Fix PHP extensions in Dockerfile

The PHP extensions in the Dockerfile have been rearranged and php${PHP_VERSION}-mbstring has been added.

cacciaresi commented on composer.json:
```diff
@@ -26,6 +26,6 @@
     "scripts": {
         "style:check": "docker-compose -f docker-compose.cli.yml run --rm php-cs-fixer fix --verbose --show-progress=estimating --dry-run",
         "style:fix": "docker-compose -f docker-compose.cli.yml run --rm php-cs-fixer fix --verbose --show-progress=estimating",
```
This will run inside the container, so I'd remove the from here and add the instructions to the readme (we need to create one):

```bash
✗  docker compose -f docker-compose.cli.yml run --rm php-cs-fixer fix --verbose --dry-run
PHP CS Fixer 3.1.0 River by Fabien Potencier and Dariusz Ruminski (cf4cedb)
Runtime: PHP 7.1.33
Loaded config default from "/data/.php-cs-fixer.php".
........                                                                                                                                                                                  8 / 8 (100%)
Legend: ?-unknown, I-invalid file syntax (file ignored), S-skipped (cached or empty file), .-no changes, F-fixed, E-error

Checked all files in 0.090 seconds, 12.000 MB memory used
```

So the readme should have a section for installing dependencies:

```bash

docker compose run --no-deps --rm php-cli composer install
```

and another to execute lint:

```bash
docker compose -f docker-compose.cli.yml run --rm php-cs-fixer fix --verbose --dry-run
```

juanariastn replied:
[e22ce50](https://github.com/TiendaNube/image-compression-lib/pull/13/commits/e22ce504d1c3f2f4337e626e6146b2ce4fc23986)

juanariastn committed:
Remove sendmail_path from PHP ini

The sendmail_path directive has been removed from PHP configuration overrides. The file php-ini-overrides-phpdbg.ini was deleted from the PHP CLI Docker setup, and the related COPY instruction has been removed from the Dockerfile.

juanariastn committed:
Add README.md

A README.md file has been created, providing library information and instructions for running lint, unit tests, and rector. In addition, unnecessary scripts in composer.json have been removed.

juanariastn commented on .docker/php-cli/Dockerfile:
```diff
@@ -1,31 +1,37 @@
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* /usr/share/doc/*
 
 # Override ini file
 COPY .docker/php-cli/php-ini-overrides.ini /etc/php/7.0/cli/conf.d/99-overrides.ini
```
[b601346](https://github.com/TiendaNube/image-compression-lib/pull/13/commits/b6013465d9426fa8ea29af706cf639478f8f1825)
